Lib/test/test_typing.py

```diff
@@ -4091,6 +4091,20 @@ class C(Generic[T]): pass
         with self.assertRaises(TypeError):
             C[()]
 
+    def test_generic_subclass_checks(self):
+        for typ in [list[int], List[int], tuple[int, str], Tuple[int, str],
+                    dict[str, int], Dict[str, int], set[str], typing.Set[str]]:
+            with self.subTest(typ=typ):
+                self.assertRaises(TypeError, issubclass, typ, object)
+                self.assertRaises(TypeError, issubclass, typ, type)
+                self.assertRaises(TypeError, issubclass, typ, typ)
```

Comment:
It may be more interesting to test `issubclass(object,  typ)`. Or it is already tested anywhere? Then perhaps that place is a better place for new tests.

Reply:
Yes, there's a single test that I am was able to find: https://github.com/python/cpython/blob/39ef93edb9802dccdb6555d4209ac2e60875a011/Lib/test/test_typing.py#L4081

It is on line `4081`, my new test is right below it.

I've added `issubclass(object, typ)` to this test as well.